src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -223,44 +227,87 @@ private ConstantValue DefaultSyntaxValue
-        // Don't always get it from the compilation because we might be in a speculative context (local function parameter),
-        // in which case the declaring compilation is the wrong one.
-        protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder binder)
+        private Binder GetBinder(SyntaxNode syntax)
+        {
+            var binder = ParameterBinderOpt;
+
+            // If binder is null, then get it from the compilation. Otherwise use the provided binder.
+            // Don't always get it from the compilation because we might be in a speculative context (local function parameter),
+            // in which case the declaring compilation is the wrong one.
+            if (binder == null)
+            {
+                var compilation = this.DeclaringCompilation;
+                var binderFactory = compilation.GetBinderFactory(syntax.SyntaxTree);
+                binder = binderFactory.GetBinder(syntax);
+            }
+            Debug.Assert(binder.GetBinder(syntax) == null);
+            return binder;
+        }
+
+        private void NullableAnalyzeParameterDefaultValueFromAttributes(DiagnosticBag diagnostics)
         {
+            var defaultValue = DefaultValueFromAttributes;
+            if (defaultValue == null || defaultValue.IsBad)
+            {
+                return;
+            }
+
             var parameterSyntax = this.CSharpSyntaxNode;
             if (parameterSyntax == null)
             {
-                return ConstantValue.NotAvailable;
+                // If there is no syntax at all for the parameter, it means we are in a situation like
+                // a property setter whose 'value' parameter has a default value from attributes.
+                // There isn't a sensible use for this in the language, so we just bail in such scenarios.
+                return;
             }
 
-            var defaultSyntax = parameterSyntax.Default;
-            if (defaultSyntax == null)
+            var binder = GetBinder(parameterSyntax);
+
+            // Nullable warnings *within* the attribute argument (such as a W-warning for `(string)null`)
+            // are reported when we nullable-analyze attribute arguments separately from here.
+            // However, this analysis of the constant value's compatibility with the parameter
+            // needs to wait until the attributes are populated on the parameter symbol.
+            var parameterEqualsValue = new BoundParameterEqualsValue(
+                parameterSyntax,
+                this,
+                ImmutableArray<LocalSymbol>.Empty,
+                // note that if the parameter type conflicts with the default value from attributes,
+                // we will just get a bad constant value above and return early.
+                new BoundLiteral(parameterSyntax, defaultValue, Type));
+
+            NullableWalker.AnalyzeIfNeeded(binder, parameterEqualsValue, diagnostics);
```

Comment:
How is this information getting to the semantic model? IE, hovering over the `default` keyword for a NRT should have a nullable state of MaybeNull, are there tests for this?

Reply:
I actually have no idea, but I will try to add some tests that I think cover this gap for this as well as for attributes which use a similar approach for nullable analysis.